sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -165,6 +165,85 @@ namespace sycl::ext::oneapi::experimental::matrix {
+
+In order to be able to perform any piece-wise operation in a general and in an efficient way, we provide a mapping conversion function from the matrix domain that is owned by a group of work items to the portion that is owned by a work item. Besides, the WI data to joint matrix mapping coordinates information must be known. 
+
+Nvidia wmma interface added a new member to `fragment` class to designate the WI owned part of the matrix. 
+While this provides fast element indexing on the GPU compared to the non-restrictive option, the user does not know the mapping of the owned data to the original matrix. 
+ However using the `mma` ptx instructions as opposed to the `wmma` ptx instructions the mapping is known. Knowing this mapping is important for the user to implement new operations like sum of rows of a matrix for quantized algorithms.
+
+##### Solution: Explicit conversion with mapping from SIMD to SPMD
+We introduce a new function `get_wi_data` that provides a view of the portion of the matrix that is owned by the current WI.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+template <typename T, size_t NumRows, size_t NumCols,
+          matrix_layout Layout = matrix_layout::row_major,
+          typename Group = sycl::sub_group>
+struct joint_matrix {
+   joint_matrix(Group sg) {…}   
+   wi_data<T, NumRows, NumCols, Layout, Group> get_wi_data() {
+     return wi_data<T, NumRows, NumCols, Layout, Group>(*this);
+   }
+};
+template <typename T, size_t NumRows, size_t NumCols, matrix_layout Layout, typename Group>
+class wi_data {
+  joint_matrix<T, NumRows, NumCols, Layout, Group> &M;
+public:
+  wi_data(joint_matrix<T, NumRows, NumCols, Layout, Group> &Mat) : M(Mat){}
+  size_t length() {return __spirv_JointMatrixWorkItemLengthINTEL(M.spvm);}
+  wi_element<T, NumRows, NumCols, Layout, Group> operator[](size_t i) {
+    return wi_element<T, NumRows, NumCols, Layout, Group>(M, i);
+  }
+};
+template <typename T, size_t NumRows, size_t NumCols,
+          matrix_layout Layout = matrix_layout::row_major,
+          typename Group = sycl::sub_group>
+class wi_element {
+  joint_matrix<T, NumRows, NumCols, Layout, Group> &M;
+  std::size_t idx;
+
+public:
+  wi_element(joint_matrix<T, NumRows, NumCols, Layout, Group> &Mat, std::size_t i): M(Mat), idx(i) {}
+  operator T() {
+    return __spirv_VectorExtractDynamic(M.spvm, idx);
+  }
+  wi_element &operator=(const T &rhs) {
+    M.spvm = __spirv_VectorInsertDynamic(M.spvm, rhs, idx);
+    return *this;
+  }
+…
+};
+}
+```
```

Comment:
After the code synopsis, there should be some description of the member functions.  I'd suggest three tables, one for each class:

* Table describing member functions of `joint_matrix` (`get_wi_data`)
* Table describing member functions of `wi_data`
* Table describing member functions of `wi_element`.

You can see an example here: https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_device_global.asciidoc#representation-of-device-globals

(Scroll down to the table after the code synopsis.)